### ISSUE TITLE
diag_resume_bimodal: a diagnostic child that dies before printing is counted GREEN

### ISSUE THREAD

**Issue by cklxx** [acupof-ai/aupai#581]
Found in de's second read of #549 (branch tip `71f82b90`, unpushed; PR head `cd4f527b` is its ancestor).

`coloc` decides the diag arm by parsing stdout, never the returncode:

```python
line = next((l for l in d.stdout.splitlines() if l.startswith("DIAGPAIR")), "DIAGPAIR ?")
d_is_red = line.startswith("DIAGPAIR RED") or line.startswith("DIAGPAIR ARM")
```

`pair_once` prints that line only on completion, so a child killed beforehand (ENOSPC / OOM / OOM killer — ordinary long-job deaths) leaves stdout empty, the fallback `"DIAGPAIR ?"` is used, both `startswith` calls fail, and the arm is counted green.

Reproduced against the real code (`subprocess.run` patched: gate rc=0, diag rc=3 with empty stdout, then `coloc(3)`'s own loop):

```
iter 0: gate=green DIAGPAIR ?
iter 1: gate=green DIAGPAIR ?
iter 2: gate=green DIAGPAIR ?
COLOCATE SUMMARY over 3 (stress=0): gate_red=0 diag_red=0 disagreements=0
```

Three dead diagnostics are output-identical to three green ones. `pair_once` already has `DIAGPAIR ARM-CRASH` for the `_run_pair`-returned-non-zero path, so only "never reached the print" leaks.

**Consequence, measured.** One pair writes 4.8 GiB (per-leaf per-step fp32 logits dumps via `torch.save`); the default out dir is hardcoded `/tmp/diag_colocate_out`. ENOSPC surfaces as a `torch.save` exception, i.e. exactly this path — so the run reports green. 8 pairs ≈ 38 GiB and the dispatch matrix's 6 ≈ 29 GiB, against ~14 GB of runner `/tmp`. The instrument needs more scratch than the thing it measures.

The job is `continue-on-error` by design, so no one goes red on this. What it breaks is the distinction the diagnostic exists for: "did not reproduce" vs "never ran", on a flake that is only visible on runners.

Fix: treat a missing `DIAGPAIR` line as red via `d.returncode != 0`, or use a `DIAGPAIR NORUN` sentinel and count it red. Separately, cap the pair count or require explicit `--out`.

**Comment by cklxx** [comment]
**Verified fixed at `bf9f0afc`** (on top of `71f82b90`, unpushed).

Ran the exact reproduction that found it — `subprocess.run` patched so the gate returns rc=0 and the diag child returns rc=3 with empty stdout, then `coloc(3)`'s real loop body:

```
iter 0: gate=green DIAGPAIR NORUN (nonzero exit before any DIAGPAIR line; rc=3)  <-- GATE/DIAG DISAGREE
iter 1: gate=green DIAGPAIR NORUN (nonzero exit before any DIAGPAIR line; rc=3)  <-- GATE/DIAG DISAGREE
iter 2: gate=green DIAGPAIR NORUN (nonzero exit before any DIAGPAIR line; rc=3)  <-- GATE/DIAG DISAGREE
COLOCATE SUMMARY over 3 (stress=0): gate_red=0 diag_red=3 disagreements=3
```

Three dead children now count red with a cause, where before they summed to `diag_red=0`. `_classify_pair` is a real three-state function and the selftest pins all four verdicts.

Disk guard also verified two-sided: at the machine's actual 0.4 GiB free it prints `COLOCATE ABORT ... need ~6 GiB for one pair` and returns **without starting a run**; with `disk_usage` patched to 50 GiB the sweep proceeds and, with each pair writing a file, `rmtree` leaves none behind — N pairs now occupy one pair's disk.

Two non-blocking cleanups reported to 0e:
1. The docstring says `GREEN is positive (rc0 AND a DIAGPAIR GREEN line)`, but the code returns green on the line alone — `_classify_pair(3, "DIAGPAIR GREEN")` and `(137, "DIAGPAIR GREEN")` both return green. Unreachable today (`pair_once` exits 0 on every path that can print GREEN), so not a live hole, but it is a criterion that states a conjunct it does not apply. Either apply rc0 or correct the sentence.
2. The `line = next(...)` at `:604-605` is now dead — `_classify_pair` overwrites it on the next line.

The two ENOSPC-related consequences are therefore both closed: the instrument no longer manufactures the death, and no longer mis-scores it if it happens anyway.

**Comment by cklxx** [comment]
Fixed on branch 0e-ci-resume-diag: bf9f0afc (_classify_pair three-state; an early-dead child with no DIAGPAIR line is now DIAGPAIR NORUN and counted red; coloc pre-checks ~6 GiB and removes per-pair tensors) and ca031c38 (docstring/dead-line cleanup). Verified with the exact monkeypatched dead-child (gate rc0, diag rc3, empty stdout): coloc(3) now reports gate_red=0 diag_red=3. Keeping this open until #549 merges to main, then it can close.
